docs/adr/0006-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,123 @@
+      * Perform the same process whenever a new beta or release application-services version is published.
+    * All application-services changes that are breaking for consumers should have corresponding PRs in the consumer repos that have been reviewed and approved.  If we accidentally merge the change into application-services before the PRs are ready, then we should back the change out.
+
+## Decision Outcome
+
+**(B) Move to a rapid release process where changes in app-services are frequently merged into consumer applications**
+
+## Pros and Cons of the Options
+
+### Keep our current release process
+
+  * Good, because it doesn't require changes to our workflow
+  * Bad, for the reasons enumerated at the top of this document
+
+### Move to a rapid release process where changes in app-services are frequently merged into consumer applications
+
+  * Good, because code will get merged and released more frequently.
+  * Good, because we don't need to spend time cutting releases.
+  * Good, because uplifts can be done much quicker.
+  * Good, because more of our release process will be automated.
+  * Bad, because we need implement new automation.
+  * Bad, because when breaking changes cause the automatic nightly merge to fail for an application.  This means a developer will get the relatively urgent task of resolving things.
+  * Good, because this avoids the current state of affairs where our `main` branch is not ready to be taken by the applications.  This means that a) our current code is not getting used and tested and b) it's very complicated to get other changes to that application, including bugfixes.
+  * Bad, because breaking changes require the corresponding PRs to be approved from all other teams before they can be merged into application-services.
+  * Good, because this is how things will work in a monorepo world.  Adjusting our workflow now prepares us for the future and will give us insights that may help guide the monorepo migration.
+  * Good, because our version numbers will align more closely with the rest of Mozilla.
+  * Good, because it makes it easier to know which application-services version the applications are using.  The current Firefox X release will always be using the latest application-services.X release -- for both the mobile apps and Desktop.
+
+# Non-normative discussion
+
+## Impact on other teams
+### Application teams -- Android, iOS, and Desktop
+
+* Phase 1
+  * The application-services version number style will change
+  * There may be a increase in the amount of application-service version upgrade PRs to approve.  We'll want to make sure that each application is on an application-services release that matches their release (i.e. avoiding having Fenix 110.x on application-services 109.x).  For the Android/iOS teams this will only be a small increase, but for Desktop it will be larger.
```

Comment:
Can we leverage updatebot for this on the Desktop side to automate most of the process? So eventually someone on the A-S team just needs to r+ a Phabricator rev and queue it for Lando?

Reply:
That sounds pretty great to me.  I'm hoping we can leverage the existing automation as much as possible.

Reply:
Feel free to resolve this again, but I also just wanted to draw your attention to the NSS process so you can see another option. They utilize a single tracking bug for every cycle (for example: https://bugzilla.mozilla.org/show_bug.cgi?id=1805486) with periodic pushes during the Nightly cycle until the final release is tagged at the end of it. For A-S, I think I'd lean towards preferring having a new bug for each upstream bump (bugs and automation are cheap after all, and it makes for better bisection down the road), but that's another option anyway.